src/pages/campaigns/CampaignEditions.tsx

```diff
@@ -0,0 +1,316 @@
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import {
+  Breadcrumb,
+  BreadcrumbItem,
+  BreadcrumbLink,
+  BreadcrumbList,
+  BreadcrumbPage,
+  BreadcrumbSeparator,
+} from "@/components/ui/breadcrumb";
+import { Button } from "@/components/ui/button";
+import { Edit, Eye, Plus, Trash2 } from "lucide-react";
+import { Badge } from "@/components/ui/badge";
+import Sidebar from "../../components/Sidebar";
+import AlertDialogComponent from "@/components/NewsletterComponents/AlertDialogComponent";
+import PreviewDialog from "@/components/NewsletterComponents/PreviewDialog";
+
+export interface campaignEditionById {
+  campaignTitle: string;
+  campaignContent: {
+    id: number;
+    content: string;
+    status: Status;
+    scheduledOn: Date;
+  }[];
+}
+
+export const tempData: campaignEditionById = {
+  campaignTitle: "The Monthly Roundup - September",
+  campaignContent: [
+    {
+      id: 1,
+      content: "In this issue, we cover our recent events.",
+      status: Status.Published,
+      scheduledOn: new Date("2022-09-10T10:00:00.000Z"),
+    },
+    {
+      id: 2,
+      content: `
+# [Newsletter Title]
+*Date: YYYY-MM-DD*
+
+---
+
+## Welcome Message
+Hello [Readers],
+
+Welcome to this month's edition of our newsletter! We have some exciting updates and resources to share with you.
+
+---
+
+## Featured Article
+### [Article Title]
+*Author: [Author Name]*  
+*Published on: YYYY-MM-DD*
+
+[Brief introduction to the article or summary. You can include a link to read more.](#)
+
+---
+
+## Upcoming Events
+### [Event Title]
+- **Date:** YYYY-MM-DD
+- **Time:** HH:MM AM/PM
+- **Location:** [Venue or Link to Event]
+  
+[Short description of the event and a link for registration if applicable.]
+
+---
+
+## Community Highlights
+- **Highlight 1:** [Description or link]
+- **Highlight 2:** [Description or link]
+- **Highlight 3:** [Description or link]
+
+---
+`,
+      status: Status.Draft,
+      scheduledOn: new Date("2023-09-15T10:00:00.000Z"),
+    },
+    {
+      id: 3,
+      content: "This month, we dive into AI advancements.",
+      status: Status.Published,
+      scheduledOn: new Date("2021-09-10T10:00:00.000Z"),
+    },
+    {
+      id: 4,
+      content: "Our top picks for the best books of the year.",
+      status: Status.Published,
+      scheduledOn: new Date("2021-09-10T10:00:00.000Z"),
+    },
+  ],
+};
+
+const CampaignEditions = () => {
+  const { id } = useParams<{ id: string }>();
+
+  const [campaignEdition, setcampaignEdition] = useRecoilState(
+    campaignEditionByIdData
+  );
+  const [isPreviewDialogOpen, setIsPreviewDialogOpen] = useState(false);
+  const [editionToPreview, setEditionToPreview] = useState<number | null>(null);
+  const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
+  const [editionToDelete, setEditionToDelete] = useState<number | null>(null);
+
+  useEffect(() => {
+    document.title = "ACE | Campaign Editions";
```

Comment:
Setting up the title happens only once. Have a useEffect with zero dependency array and execute it there so that it only gets executed upon screen load.

Reply:
Done